src/Raven.Embedded/EmbeddedServer.cs

```diff
@@ -104,6 +111,12 @@ public void StartServer(ServerOptions? options = null)
             GC.KeepAlive(task); // make sure that we aren't allowing to elide the call
         }
 
+        // Solution 2: Add a STop method that allow graceful termination AND forceful termination via cancellation
+        public async Task StopAsync(CancellationToken cancellationToken)
+        {
+            // Run code from the Dispose and ensure ShutdownServerProcess is invoked with a cancellation token
```

Comment:
What is the meaning of that here? That you start graceful and if cancelled, you kill the process?

Reply:
Yes, indeed, this is the pattern that cooperative cancellation applies. When the method is invoked, just run gracefully until the caller/invoker cancels and then cooperate with that request and continue with an ungraceful termination.